cpp/src/decisiontree/decisiontree.cu

```diff
@@ -55,8 +56,7 @@ void set_tree_params(DecisionTreeParams &params, int cfg_max_depth,
  * @param[in] params: decision tree hyper-parameters.
  */
 void validity_check(const DecisionTreeParams params) {
-  ASSERT((params.max_depth == -1) || (params.max_depth > 0),
-         "Invalid max depth %d", params.max_depth);
+  ASSERT((params.max_depth > 0), "Invalid max depth %d", params.max_depth);
```

Comment:
This means, users now MUST pass the `max_depth` param to DT/RF? Why so?

Reply:
Because its a for loop using depth. passing maxdepth is compulsory.